src/oaklib/implementations/gilda.py

```diff
@@ -0,0 +1,39 @@
+"""A text annotator based on Gilda."""
+
+from dataclasses import dataclass
+from typing import Iterator
+
+from oaklib.datamodels.text_annotator import TextAnnotation, TextAnnotationConfiguration
+from oaklib.interfaces import TextAnnotatorInterface
+from oaklib.interfaces.text_annotator_interface import TEXT, nen_annotation
+
+__all__ = [
+    "GildaImplementation",
+]
+
+
+@dataclass
+class GildaImplementation(TextAnnotatorInterface):
+    """Perform named entity normalization on text strings with Gilda [gyori2021]_.
+
+    .. [gyori2021] Benjamin M Gyori, Charles Tapley Hoyt, Albert Steppi (2021)
+        `Gilda: biomedical entity text normalization with machine-learned
+        disambiguation as a service <https://doi.org/10.1093/bioadv/vbac034>`_,
+        *Bioinformatics Advances*, Volume 2, Issue 1, 2022, vbac034,
+    """
+
+    def annotate_text(
+        self, text: TEXT, configuration: TextAnnotationConfiguration = None
+    ) -> Iterator[TextAnnotation]:
+        if not configuration:
+            raise NotImplementedError
+        if not configuration.matches_whole_text:
+            raise NotImplementedError
+
+        import gilda
```

Comment:
Wondering about this. How would this work when we want to add a unit test for this?

My preference would be either:

1. have a separate gilda plugin (preliminary docs here file:///Users/cjm/repos/ontology-access-kit/docs/_build/html/howtos/write-a-plugin.html)
2. add this to pyproject.toml + poetry.lock (maybe that is the ultimate the intent and you just didn't want to risk commit conflicts for now?)

Reply:
I really want to stress that plug-ins should only be used by external people who want to extend the package but not in an open source way - eg the million different repo approach isn’t sustainable (I learned the hard way in bio2bel)

you can always have the testing harness install optional dependencies and we can also make it a bit easier from this class to mock the Gilda grounder. I will provide a small unit test tomorrow